src/logicLayer/GameLogic.java

```diff
@@ -70,11 +74,69 @@ private boolean checkIfUnitIsOnTile(Tile tile) {
 		return tile.hasUnit();
 	}
 	
-	private int calculateMovCostBetweenTile(Tile startTile, Tile finalTile) {
-		int xDiff = Math.abs(startTile.getPos().getXPosition() - finalTile.getPos().getXPosition());
-		int yDiff = Math.abs(startTile.getPos().getYPosition() - finalTile.getPos().getYPosition());
-		int moveCost = xDiff+yDiff;
-		return moveCost;
+	private List<Tile> calculateValidTileToMoveTo(Unit unit) {
+		List<Tile> tileRange = new ArrayList<Tile>();
+		Direction direction = Direction.NONE;
+		Tile startTile = unit.getTile();	
+		int movStamina = unit.getStats().getMov().getCurrentValue()+1;
+		tileRange.add(startTile);
+		calculateMovRange(tileRange, startTile, movStamina, direction);
+		System.out.println();
+		return tileRange;
+	}
+	
+	private void calculateMovRange(List<Tile> tileRange, Tile startTile, int movStamina, Direction direction) {
+		if (movStamina == 0) {
+		}
+		else {
+			movStamina--;
+			if (direction.equals(Direction.NONE)) {
+				calculateMovRange(tileRange, startTile, movStamina, Direction.UP);
+				calculateMovRange(tileRange, startTile, movStamina, Direction.RIGHT);
+				calculateMovRange(tileRange, startTile, movStamina, Direction.DOWN);
+				calculateMovRange(tileRange, startTile, movStamina, Direction.LEFT);
+			}
+			if (direction.equals(Direction.RIGHT)) {
+				Tile newTile= gameMap.getTileAtCoordinates(startTile.getPos().getYPosition()+1,
+						startTile.getPos().getXPosition());
+				if (newTile != null) {
+					tileRange.add(newTile);
+					calculateMovRange(tileRange, newTile, movStamina, Direction.UP);
+					calculateMovRange(tileRange, newTile, movStamina, Direction.RIGHT);
+					calculateMovRange(tileRange, newTile, movStamina, Direction.DOWN);	

```

Comment:
If I understand correctly, this will add all movable tiles to a list.
I see that for each direction, it won't consider moving backwards, thus counting the same tile twice.
But won't it still count in a circle? Like it would start on one particular tile, but eventually it would count the one above, then to the right, then down, then left, thus counting the starting tile twice.

Let me know if I worded that confusingly, or if this even matters at all

Reply:
This is a feature not a bug. There could be many ways to get to the same tile (in the case you raised the start tile) but each way may not have the same move cost associated. As such, in order to get an accurate way of getting the movable tiles, we need to consider the different ptaths we can take.